### PR TITLE
Nerfs flare and glowstick burn durations

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request

Who the fuck made flares burn for 35 minutes. These things are never gonna burn out, people will just assume they last forever.

Flares now burn for between 10 and 15 minutes down from 25 to 35. Nerfs glowsticks too, fuckin 60 MINUTES WHAT, from 50/60 to 20/25
This should be enough time to make proper use of them (particularly for the crashed miner) but not so long they feel everlasting

Glowsticks also hide their action button when they are lit (since they can't really be unlit)

Oh and glowsticks actually burn now. The old code assumed attack_hand would return TRUE if it did anything, but it just... didn't. 
Let's deal with that.

## Why It's Good For The Game

Flares and glowsticks will feel more like flares and glowsticks (I hope)
Also I'll have an excuse to make them brighter (coming soon)

## Changelog
:cl:
balance: Nerfs the burn times of flares and glowsticks, from 25/35 to 10/15 and 50/60! to 20/25. 
balance: Glowsticks now dim as they burn
add: Glowsticks no longer leave an unusable action button sitting around then they're lit
fix: Glowsticks actually burn down now (The code was broken)
/:cl:

### CODE REVIEW

**Comment by jughu** [comment]
my poor candles getting a balance patch :(
would be cooler if you could see the candle sink every 5 minutes so you could see how long your candle is gonna last (or atleast assume)

**Comment by Jackal-boop** [comment]
Candles don't emit light so I think it's fine for them to burn so long

**Comment by Xkallubar** [comment]
No.

**Comment by LemonInTheDark** [comment]
> Candles don't emit light so I think it's fine for them to burn so long

They do is the thing, it's just weak and crummy (see #81425)

> my poor candles getting a balance patch :( would be cooler if you could see the candle sink every 5 minutes so you could see how long your candle is gonna last (or atleast assume)

They do actually have bespoke icon states for this, I need to update the time constants there to match the new bounds tho. (The fact you didn't know this I think speaks to how long this takes)

image

> No.

Elaborate on your concerns challenge

**Comment by deathrobotpunch** [comment]
is this really necessary? it’s not like using flares is the new meta strat or anything

**Comment by wesoda25** [comment]
Candles are pretty much purely aesthetic, personally I'd prefer to see them stay at 35. I enjoy when they finally reach the half melted stage and how long they stay there for (which afaik isn't a possibility with eternal candles which... afaik is a thing?)

Ultimately whatever tho and flare/glowstick change is certainly fair

**Comment by LemonInTheDark** [comment]
> Candles are pretty much purely aesthetic, personally I'd prefer to see them stay at 35. I enjoy when they finally reach the half melted stage and how long they stay there for (which afaik isn't a possibility with eternal candles which... afaik is a thing?)

My concern is almost entirely an aesthetic one. Like, irl glowsticks last like 8 hours, but when you snap a glowstick or light a flare, you're expecting something temporary, and I'd like the span of time it takes to MAKE it temporary to be more then an hour.

Applying this to candles it a bit less strong because they weren't THAT far out there. I think I'll just send em back

**Comment by deathrobotpunch** [comment]
could you give survivors a more permanent light source then?

**Comment by LemonInTheDark** [comment]
> could you give survivors a more permanent light source then?

I was wrong he doesn't really have a reason to use it, since he gets glowing plants. It's slightly better but not like, THAT much better.

**Comment by deathrobotpunch** [comment]
> I was wrong he doesn't really have a reason to use it, since he gets glowing plants. It's slightly better but not like, THAT much better.

you alright? you seem to be talking to people that don’t exist

**Comment by LemonInTheDark** [comment]
I was responding to you in context of my misremembering of the survivalist's lighting situation, unsure what you're talkin bout

File: code/game/objects/items/devices/flashlight.dm [hunk 1]
```diff
@@ -415,7 +415,7 @@
 /obj/item/flashlight/flare/Initialize(mapload)
 	. = ..()
 	if(randomize_fuel)
-		fuel = rand(25 MINUTES, 35 MINUTES)
+		fuel = rand(5 MINUTES, 10 MINUTES)
```
Comment (Ghommie) [inline code/game/objects/items/devices/flashlight.dm:418]: I think they should be a bit more consistent, the range is too wide here. Also, it should at least burn twice as bright if it were to last less than half as long.

File: code/game/objects/items/devices/flashlight.dm [hunk 2]
```diff
@@ -762,7 +762,7 @@
 	var/fuel = 0
 
 /obj/item/flashlight/glowstick/Initialize(mapload)
-	fuel = rand(50 MINUTES, 60 MINUTES)
+	fuel = rand(10 MINUTES, 15 MINUTES)
```
Comment (Ghommie) [inline code/game/objects/items/devices/flashlight.dm:765]: Maybe 18-20 minutes, or 20-25 if you feel like spending a couple minutes making it so their range wanes with time.

**Review by Ghommie — CHANGES_REQUESTED** [review]

**Comment by bigfatbananacyclops** [comment]
Bro the long time flare lights up the game is my cup of tea and for many miner... Are we making a survival game?

File: code/game/objects/items/devices/flashlight.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/game/objects/items/devices/flashlight.dm:418]: I'm changing the lighting w https://github.com/tgstation/tgstation/pull/81425 which makes this a bit harder.
"twice as bright" isn't really meaningful at some point cause we don't have bloom, so we just end up blowing out the color on the light.

I will see about the range tho

File: code/game/objects/items/devices/flashlight.dm (hunk 2 above)
Comment (LemonInTheDark) [reply code/game/objects/items/devices/flashlight.dm:765]: I'll buy that, sure

**Pull request merged**